systemli/ticker
# ✨ Add a Signal group bridge

## Summary

* Creates a Signal group when the bridge is configured, with following settings:
  - name & description as configured
  - predefined avatar (file stored in the signal-cli container)
  - invite link enabled
  - add member permissions: every member
  - edit details permissions: only admin
  - send messages: only admin
* Sends messages (with attachments) to the Signal group
* Tries to leave the Signal group when bridge is reset. This doesnt work though if the Signal account is the only admin (which is almost always the case).
* Tries to delete message when it's deleted in ticker.

## How to test

1. Start [signal-cli](https://github.com/AsamK/signal-cli/).
2. Register a Signal account with `signal-cli` or copy an existing one to the `data/` directory.
3. Configure the Signal bridge in `config.yml`:
   ```yaml
   signal_group:
     api_url: "http://signal-cli.example.org/api/v1/rpc"
     account: "+***********"
   ```
4. [optional] Copy avatar file `ticker.svg` to `data/ticker.svg`
5. Start ticker and configure Signal bridge through API call:
   ```sh
   $ curl -X PUT -H "Accept: application/json" -H "Content-Type: application/json" -H "Authorization: Bearer [...]" --data '{"active":true,"groupName":"Ticker Test Group","groupDescription":"Ticker Group"}' http://localhost:8080/v1/admin/tickers/1/signal_group
   ```
6. Join the Signal group via invite link with another account
7. Send messages (with attachments) via ticker-admin and see them being sent to the Signal group
